native/src/main/java/io/ballerina/stdlib/rabbitmq/util/ChannelUtils.java

```diff
@@ -97,20 +98,36 @@ public static Object queueDeclare(Environment environment, BObject clientObj,
                 exclusive = config.getBooleanValue(RabbitMQConstants.QUEUE_EXCLUSIVE);
                 autoDelete = config.getBooleanValue(RabbitMQConstants.QUEUE_AUTO_DELETE);
                 if (config.getMapValue(RabbitMQConstants.QUEUE_ARGUMENTS) != null) {
-                    argumentsMap = (HashMap<String, Object>) config.getMapValue(RabbitMQConstants.QUEUE_ARGUMENTS);
+                    @SuppressWarnings(RabbitMQConstants.UNCHECKED)
+                    HashMap<BString, Object> queueArgs =
+                            (HashMap<BString, Object>) config.getMapValue(RabbitMQConstants.QUEUE_ARGUMENTS);
+                    queueArgs.forEach((k, v) -> argumentsMap.put(k.getValue(), getConvertedValue(v)));
                 }
             }
             channel.queueDeclare(queueName.getValue(), durable, exclusive, autoDelete, argumentsMap);
             RabbitMQMetricsUtil.reportNewQueue(channel, queueName.getValue());
             RabbitMQTracingUtil.traceQueueResourceInvocation(channel, queueName.getValue(), environment);
-        } catch (IOException | ShutdownSignalException exception) {
+        } catch (IOException | ShutdownSignalException | BError exception) {
             RabbitMQMetricsUtil.reportError(channel, RabbitMQObservabilityConstants.ERROR_TYPE_QUEUE_DECLARE);
             return RabbitMQUtils.returnErrorValue("Error occurred while declaring the queue: "
                                                           + exception.getMessage());
         }
         return null;
     }
 
+    private static Object getConvertedValue(Object v) {
+        if (v instanceof BString) {
+            return ((BString) v).getValue();
+        } else if (v instanceof BDecimal) {
+            return ((BDecimal) v).decimalValue();
```

Comment:
Does this map accepts BigDecimal values?

Reply:
`BDecimal` not BigDecimal though

Reply:
`decimalValue()` returns a `BigDecimal` value.